clients/java/src/test/java/io/zeebe/client/ZeebeClientTest.java

```diff
@@ -62,4 +68,60 @@ public void shouldGetHealthCheck() throws InterruptedException {
                       });
             });
   }
+
+  @Test
+  public void shouldDeployWorkflow() throws URISyntaxException {
+    final String filePath =
+        getClass().getResource("/workflows/demo-process.bpmn").toURI().getPath();
+
+    final DeploymentEvent deploymentResponse =
+        client.workflowClient().newDeployCommand().addResourceFile(filePath).send().join();
+
+    assertThat(deploymentResponse).isNotNull();
+    assertThat(deploymentResponse.getState()).isEqualTo(CREATED);
+    assertThat(deploymentResponse.getDeployedWorkflows().size()).isEqualTo(1);
+    assertThat(deploymentResponse.getDeployedWorkflows().get(0).getBpmnProcessId())
+        .isEqualTo("demoProcess");
+    assertThat(deploymentResponse.getDeployedWorkflows().get(0).getVersion()).isEqualTo(1);
+    assertThat(deploymentResponse.getDeployedWorkflows().get(0).getWorkflowKey()).isEqualTo(1);
+    assertThat(deploymentResponse.getDeployedWorkflows().get(0).getResourceName())
+        .endsWith("demo-process.bpmn");
+  }
+
+  @Test
+  public void shouldDeployMultipleWorkflows() throws URISyntaxException {
+    final String demoProcessPath =
+        getClass().getResource("/workflows/demo-process.bpmn").toURI().getPath();
+    final String anotherDemoProcessPath =
+        getClass().getResource("/workflows/another-demo-process.bpmn").toURI().getPath();
+    final String simpleWorkflowPath =
+        getClass().getResource("/workflows/simple-workflow.yaml").toURI().getPath();
+
+    final String[] ids = {"yaml-workflow", "anotherDemoProcess", "demoProcess"};
+    final List<String> processIds = Arrays.asList(ids);
+    final Long[] keys = {1L, 2L, 3L};
+    final List<Long> workflowKeys = Arrays.asList(keys);
+
+    final DeploymentEvent deploymentResponse =
+        client
+            .workflowClient()
+            .newDeployCommand()
+            .addResourceFile(demoProcessPath)
+            .addResourceFile(anotherDemoProcessPath)
+            .addResourceFile(simpleWorkflowPath)
+            .send()
+            .join();
+
+    assertThat(deploymentResponse).isNotNull();
+    assertThat(deploymentResponse.getDeployedWorkflows().size()).isEqualTo(3);
+
+    deploymentResponse
+        .getDeployedWorkflows()
```

Comment:
You could also do this (no need to create the separate lists):

```java
    assertThat(deploymentResponse.getDeployedWorkflows())
        .extracting(w -> w.getBpmnProcessId())
        .containsExactly(ids);
    assertThat(deploymentResponse.getDeployedWorkflows())
      .extracting(w -> w.getVersion())
      .containsOnly(1);
    assertThat(deploymentResponse.getDeployedWorkflows())
      .extracting(w -> w.getWorkflowKey())
      .containsOnly(keys);
```

But I think this is a matter of taste.

Reply:
Hm I'm not sure didn't benchmarked it.